sc_cost_meter/utils.py

```diff
@@ -0,0 +1,156 @@
+import awspricing
+import boto3
+import logging
+import os
+
+from datetime import datetime
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+os.environ["AWSPRICING_USE_CACHE"] = "1"  # enable cache for awspricing
+
+def get_ec2_client():
+  return boto3.client('ec2')
+
+def get_ssm_client():
+  return boto3.client('ssm')
+
+def get_meteringmarketplace_client():
+  return boto3.client('meteringmarketplace')
+
+def get_instances(states):
+  '''
+  Get all EC2 instances in the specfied states
+  :param states: list of states (['pending'|'running'|'shutting-down'|'terminated'|'stopping'|'stopped'])
+         https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceState.html
+  :return: list of instances in the state specified by states, empty list if none are
+           found in the specified states
+  '''
+  client = get_ec2_client()
+  response = client.describe_instances(
+    Filters=[
+       {
+          'Name': 'instance-state-name',
+          'Values': states
+       }
+    ]
+  )
+
+  instances = []
+  reservations = response["Reservations"]
+  if reservations:
+    for reservation in reservations:
+        for instance in reservation["Instances"]:
+          instances.append(instance)
+
+  return instances
+
+def get_ec2_on_demand_pricing(instance):
+  '''
+  Get the on demand pricing for an instance
+  :param instance: an EC2 in the AWS account
+  :return: on demand price of the instance (https://aws.amazon.com/ec2/pricing/on-demand/)
```

Comment:
More precisely it is the "hourly_price", not "price".

Reply:
fixed